test/wasi/test-wasi.js

```diff
@@ -0,0 +1,84 @@
+'use strict';
+const common = require('../common');
+
+if (process.argv[2] === 'wasi-child') {
+  const fixtures = require('../common/fixtures');
+  const tmpdir = require('../../test/common/tmpdir');
+  const fs = require('fs');
+  const path = require('path');
+
+  common.expectWarning('ExperimentalWarning',
+                       'WASI is an experimental feature. This feature could ' +
+                       'change at any time');
+
+  const { WASI } = require('wasi');
+  tmpdir.refresh();
+  const wasmDir = path.join(__dirname, 'wasm');
+  const wasi = new WASI({
+    args: [],
+    env: process.env,
+    preopens: {
+      '/sandbox': fixtures.path('wasi'),
+      '/tmp': tmpdir.path
+    }
+  });
+  const importObject = { wasi_unstable: wasi.wasiImport };
+  const modulePath = path.join(wasmDir, `${process.argv[3]}.wasm`);
+  const buffer = fs.readFileSync(modulePath);
+
+  (async () => {
+    const { instance } = await WebAssembly.instantiate(buffer, importObject);
+
+    wasi.start(instance);
+  })();
+} else {
+  const assert = require('assert');
+  const cp = require('child_process');
+  const { EOL } = require('os');
+
+  function runWASI(options) {
+    console.log('executing', options.test);
+    const opts = { env: { ...process.env, NODE_DEBUG_NATIVE: 'wasi' } };
+
+    if (options.stdin !== undefined)
+      opts.input = options.stdin;
+
+    const child = cp.spawnSync(process.execPath, [
+      '--experimental-wasi-unstable-preview0',
+      '--experimental-wasm-bigint',
+      __filename,
+      'wasi-child',
+      options.test
+    ], opts);
+    console.log(child.stderr.toString());
+    assert.strictEqual(child.status, options.exitCode || 0);
+    assert.strictEqual(child.signal, null);
+    assert.strictEqual(child.stdout.toString(), options.stdout || '');
+  }
+
+  runWASI({ test: 'cant_dotdot' });
+  runWASI({ test: 'clock_getres' });
+  runWASI({ test: 'exitcode', exitCode: 120 });
```

Comment:
any reason you removed the stdin/stdout/exitcode from the file itself?

Reply:
Not really, I just thought it was simpler to have the values as part of the JS tests vs. parsing values out of the C file.